cipher-symmetric/src/test/scala/tsec/SymmetricSpec.scala

```diff
@@ -78,71 +82,73 @@ class SymmetricSpec extends TestSpec with MustMatchers with PropertyChecks {
       forAll { (testMessage: String) =>
         val testPlainText = PlainText(testMessage.utf8Bytes)
-        val testEncryptionDecryption: Either[CipherError, String] = for {
-          key       <- keyGen.generateKey()
-          instance  <- JCAAEADImpure[A, M, P]
-          encrypted <- instance.encrypt(testPlainText, key)
+        val testEncryptionDecryption: IO[String] = for {
+          key       <- keyGen.generateLift[IO]
+          encrypted <- algebra.encrypt[IO](testPlainText, key)
           keyRepr = key.getEncoded
-          built     <- keyGen.buildKey(keyRepr)
-          decrypted <- instance.decrypt(encrypted, built)
-        } yield decrypted.content.toUtf8String
-        testEncryptionDecryption must equal(Right(testMessage))
+          built     <- keyGen.buildAndLift[IO](keyRepr)
+          decrypted <- algebra.decrypt[IO](encrypted, built)
+        } yield decrypted.toUtf8String
+        testEncryptionDecryption.attempt.unsafeRunSync() must equal(Right(testMessage))
       }
     }
 
     it should "Encrypt and decrypt for the same key and AEAD" in {
       forAll { (testMessage: String, aadData: String) =>
         val testPlainText = PlainText(testMessage.utf8Bytes)
         val aad           = AAD(aadData.utf8Bytes)
-        val testEncryptionDecryption: Either[CipherError, String] = for {
-          key       <- keyGen.generateKey()
-          instance  <- JCAAEADImpure[A, M, P]
-          encrypted <- instance.encryptAAD(testPlainText, key, aad)
-          decrypted <- instance.decryptAAD(encrypted, key, aad)
-        } yield decrypted.content.toUtf8String
-        testEncryptionDecryption must equal(Right(testMessage))
+        val testEncryptionDecryption: IO[String] = for {
+          key       <- keyGen.generateLift[IO]
+          encrypted <- algebra.encryptWithAAD[IO](testPlainText, key, aad)
+          decrypted <- algebra.decryptWithAAD[IO](encrypted, key, aad)
+        } yield decrypted.toUtf8String
+        testEncryptionDecryption.attempt.unsafeRunSync() must equal(Right(testMessage))
       }
     }
 
     it should "not decrypt properly for an incorrect key" in {
       forAll { (testMessage: String) =>
         val testPlainText = PlainText(testMessage.utf8Bytes)
-        val testEncryptionDecryption: Either[CipherError, String] = for {
-          key1      <- keyGen.generateKey()
-          key2      <- keyGen.generateKey()
-          instance  <- JCAAEADImpure[A, M, P]
-          encrypted <- instance.encrypt(testPlainText, key1)
-          decrypted <- instance.decrypt(encrypted, key2)
-        } yield new String(decrypted.content, "UTF-8")
+        val testEncryptionDecryption: IO[String] = for {
+          key1      <- keyGen.generateLift[IO]
+          key2      <- keyGen.generateLift[IO]
+          encrypted <- algebra.encrypt[IO](testPlainText, key1)
+          decrypted <- algebra.decrypt[IO](encrypted, key2)
+        } yield new String(decrypted, "UTF-8")
```

Comment:
`.toUtf8String` instead of `new String(decrypted, "UTF-8")`

Reply:
didn't notice, this is a leftover from writing a test before `toUtf8String` even existed.